providers/gitlab.go

```diff
@@ -34,28 +39,89 @@ func NewGitLabProvider(p *ProviderData) *GitLabProvider {
 		p.ValidateURL = &url.URL{
 			Scheme: "https",
 			Host:   "gitlab.com",
-			Path:   "/api/v4/user",
+			Path:   "/api/v4",
 		}
 	}
-	if p.Scope == "" {
-		p.Scope = "read_user"
-	}
+	p.Scope += " read_user"
```

Comment:
Perhaps we should check if `p.Scope` contains `read_user` before appending it?

Reply:
Done